common/types/list.go

```diff
@@ -210,27 +240,41 @@ type concatList struct {
 	nextList traits.Lister
 }
 
+// Add implements the traits.Adder interface method.
 func (l *concatList) Add(other ref.Val) ref.Val {
-	if other.Type() != ListType {
+	otherList, ok := other.(traits.Lister)
+	if !ok {
 		return ValOrErr(other, "no such overload")
 	}
 	if l.Size() == IntZero {
 		return other
 	}
-	if other.(traits.Sizer).Size() == IntZero {
+	if otherList.Size() == IntZero {
 		return l
 	}
 	return &concatList{
 		TypeAdapter: l.TypeAdapter,
 		prevList:    l,
-		nextList:    other.(traits.Lister)}
+		nextList:    otherList}
 }
 
+// Contains implments the traits.Container interface method.
 func (l *concatList) Contains(elem ref.Val) ref.Val {
-	return Bool(l.prevList.Contains(elem) == True ||
-		l.nextList.Contains(elem) == True)
+	prev := l.prevList.Contains(elem)
```

Comment:
Can you add a comment here to explain the reuse of the unknown check or check unknown here again?  Either is good. Thank you :D

Reply:
Sure thing. :)